src/main/java/org/truffleruby/core/regexp/RegexpOptions.java

```diff
@@ -153,7 +153,7 @@ public Encoding setup(RubyContext runtime) {
             return EUCJPEncoding.INSTANCE;
         } else if (explicitKCode == KCode.SJIS) {
             setFixed(true);
-            return Layouts.ENCODING.getEncoding(runtime.getEncodingManager().getRubyEncoding(WINDOWS31J));
+            return EncodingManager.getEncoding(WINDOWS31J);
```

Comment:
If you extract it to a method named `getWindows31jEncoding()`, will it still load even if the branch is not taken? I don't know much about Java loading but I think it loads eagerly only classes referenced  in the current method.